aiida/orm/computers.py

```diff
@@ -463,11 +477,28 @@ def set_default_mpiprocs_per_machine(self, def_cpus_per_machine: Optional[int])
         """
         if def_cpus_per_machine is None:
             self.delete_property('default_mpiprocs_per_machine', raise_exception=False)
-        else:
-            if not isinstance(def_cpus_per_machine, int):
-                raise TypeError('def_cpus_per_machine must be an integer (or None)')
+        elif not isinstance(def_cpus_per_machine, int):
+            raise TypeError('def_cpus_per_machine must be an integer (or None)')
         self.set_property('default_mpiprocs_per_machine', def_cpus_per_machine)
 
+    def get_default_memory_per_machine(self) -> Optional[int]:
+        """
+        Return the default amount of memory (kB) per machine (node) for this computer,
+        or None if it was not set.
+        """
+        return self.get_property('default_memory_per_machine', None)
+
+    def set_default_memory_per_machine(self, def_memory_per_machine: Optional[int]) -> None:
+        """
+        Set the default amount of memory (kB) per machine (node) for this computer.
+        Accepts None if you do not want to set this value.
+        """
+        if def_memory_per_machine is None:
+            self.delete_property('default_memory_per_machine', raise_exception=False)
+        elif not isinstance(def_memory_per_machine, int):
```

Comment:
I think it would be good to do the validation directly here. The same goes for the other `set_` methods. See suggestion https://github.com/aiidateam/aiida-core/issues/5257.

Reply:
Yes, I am not saying to not do the validation here, I am just saying to put the validation in one place (let's say a separate method) and have this method and the `validate` method both call that validation. But then we are sure it is consistent